### DISCUSSION TITLE
Display an Object directly in JSX without looping

### DISCUSSION THREAD

**Discussion by dickensas** [solidjs/solid#1466] [Q&A] [answered]
I understand the below technique looping though an array using `For` tag
```
const [data, setList] = createSignal([{name: "Bobby", age: 5}]);

<For each={data()}>
      {(e, i) => (
        <div>
            { e.name }
        </div>
      )}
</For>

```

what if the data is a non-array `Object` type
```
const [data, setList] = createSignal({name: "Bobby", age: 5});

<div>
    { data.name }
</div>

```
is the above code possible in `solid-js` without creating a loop?

and using `.` dot operator directly

**Comment by atk** [comment]
You would need to use `data().name` or use `createStore` for fine-grained reactivity, that is, if only some properties change at a time.

**Comment by dickensas** [comment]
`data().name` works fine with a straight forward scenario, but if it is coming from an ajax request and bound with `createResource` it is not rendering the entire component

At the same time, I am also trying to render it inside a React component, may be that could be a mistake

**Comment by atk** [comment, marked as answer]
You can set an initial value for the resource or wrap the rendering in Suspense to make sure it is already available. What do you mean, you are rendering it inside a react component? Are you trying to use solid directly in react?

**Comment by dickensas** [comment]
even if I write pure solid as below
```
import { createSignal, createResource, Show } from "solid-js";
import { render } from "solid-js/web";

const fetchUser = async (id) =>
  (await fetch(`http://localhost:5000/test`)).json();

const App = () => {
  const [userId, buttonClick] = createSignal();
  const [user] = createResource(userId, fetchUser);

  return (
    <>
	  <button onClick={() => buttonClick()} >Test</button>
      <span>{user.loading && "Loading..."}</span>
      <div>
        <pre>{user().name}</pre>
      </div>
    </>
  );
};

render(App, document.getElementById("root"));

```

I am getting below error

```
Uncaught TypeError: Cannot read properties of undefined (reading 'name')
```

It seems the object `user()` still not ready

**Reply by atk** [reply]
As I said, use an initial value for the resource:

```js
createResource(userId, fetchUser, { initialValue: { name: "" } })
```

Alternatively, you can wrap the whole thing in Suspense.

**Comment by dickensas** [comment]
Thank you I also got an example for it https://www.solidjs.com/tutorial/async_suspense
